django/forms/widgets.py

```diff
@@ -372,7 +372,10 @@ def is_initial(self, value):
         """
         Return whether value is considered to be initial value.
         """
-        return bool(value and hasattr(value, 'url'))
+        try:
+            return bool(value and getattr(value, 'url'))
```

Comment:
For simplicity, my instinct would be to revert `hasattr` when dropping Python 2 support. If you agree, could you add a comment about it?